### ISSUE TITLE
[🐛] Android: Application not responding after selecting many videos

### ISSUE THREAD

**Issue by renchap** [react-native-image-picker/react-native-image-picker#2138]
### Description

When selecting many videos in the new Android 13 picker, an Android "Application Not Responding" message appears.

If "wait" is pressed, then the application works fine, so the main thread is not the one that is stuck.

Detail about the videos is shown, and thumbnails are here as expected.

### How to repeat issue and example

- launch the example app on an Android 13 phone
- tap "select image or video"
- select 15 videos from your library
- wait for the videos to be process, then see that an ANR dialog is shown

### Solution

There should not be any ANR

### Additional Information

- Image Picker version: 5.4.0 (also 5.3.1)
- React Native version: 0.71.2 (example app), 0.71.8 (my app)
- Platform: Android
- Development Operating System: macOS

**Comment by renchap** [comment]
I think this is fixed by #2139, I am no longer reproducing the issue with this patch.

**Comment by renchap** [comment]
Update: I have been testing this further, and while the issue occurs less often, I still get ANRs when selecting a lot of videos using the new Android 13 photo picker.

I am starting to suspect this is an Android issue.

**Comment by renchap** [comment]
Still investigating this…

It only happens when `getAppSpecificStorageUri()` is called, meaning the files are copied to the app's cache directory.

This led to #2141 which fixes some issues, but unfortunately I can still reproduce the crashes.

I tried to not copy the files to the app's cache directory, but instead require Android to make the `content://` URLs persistant, and this seems to solve the issue. I will tidy up my work on this and submit a PR.

**Comment by renchap** [comment]
Another update:
- `MediaMetadataRetriever` also needs to be closed after use, I updated #2141 to do it. This change made a big difference and after some testing I did not get ANRs once the files were processed
- The processing is done in the main thread, which can cause it to be unresponsive and trigger ANRs while the files are processed after being picked. This also makes the app UI non-responsive. Opened #2142 to run the processing in another thread

With those 2 changes I am no longer experiencing ANRs or crashed, even when selecting 50 videos on the phone.
`getAppSpecificStorageUri()` still makes things quite slow as every file needs to be copied in the app's storage, so I may still send yet another PR to allow avoiding this step through an option.
